watcher/tool/watcher/main.go

```diff
@@ -17,46 +17,101 @@ limitations under the License.
 package main
 
 import (
-	"flag"
 	"fmt"
 	"os"
 
 	"github.com/gravitational/monitoring-app/watcher/lib/constants"
+	"github.com/gravitational/monitoring-app/watcher/lib/influxdb"
 	"github.com/gravitational/monitoring-app/watcher/lib/kubernetes"
+	"github.com/sirupsen/logrus"
 
 	"github.com/gravitational/trace"
-	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
 )
 
+var (
+	mode           string
+	debug          bool
+	influxDBConfig influxdb.Config
+
+	envs = map[string]string{
+		"INFLUXDB_ADMIN_USERNAME":    "influxdb-admin-username",
+		"INFLUXDB_ADMIN_PASSWORD":    "influxdb-admin-password",
+		"INFLUXDB_GRAFANA_USERNAME":  "influxdb-grafana-username",
+		"INFLUXDB_GRAFANA_PASSWORD":  "influxdb-grafana-password",
+		"INFLUXDB_TELEGRAF_USERNAME": "influxdb-telegraf-username",
+		"INFLUXDB_TELEGRAF_PASSWORD": "influxdb-telegraf-password",
+	}
+
+	rootCmd = &cobra.Command{
+		Use:   "watcher",
+		Short: "Utility to manage InfluxDB/Grafana/Alerts",
+		RunE:  root,
+	}
+)
+
+func init() {
+	level := logrus.InfoLevel
+
+	rootCmd.PersistentFlags().StringVar(&mode, "mode", "", fmt.Sprintf("Watcher mode: %v", constants.AllModes))
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBAdminUser, "influxdb-admin-username", constants.InfluxDBAdminUser, "InfluxDB administrator username")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBAdminPassword, "influxdb-admin-password", constants.InfluxDBAdminUser, "InfluxDB administrator password")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBGrafanaUser, "influxdb-grafana-username", constants.InfluxDBGrafanaUser, "InfluxDB grafana username")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBGrafanaPassword, "influxdb-grafana-password", constants.InfluxDBGrafanaUser, "InfluxDB grafana password")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBTelegrafUser, "influxdb-telegraf-username", constants.InfluxDBTelegrafUser, "InfluxDB telegraf username")
+	rootCmd.PersistentFlags().StringVar(&influxDBConfig.InfluxDBTelegrafPassword, "influxdb-telegraf-password", constants.InfluxDBTelegrafUser, "InfluxDB telegraf password")
+	rootCmd.PersistentFlags().BoolVar(&debug, "debug", false, "Debugging mode")
```

Comment:
Usually this is called `Increase logging verbosity` as `debugging mode` is too vague imho.